lib/public_ip/service/what_is_my_ip.rb

```diff
@@ -12,7 +12,8 @@ def self.headers
       end
 
       def self.parse_ip_address(response_body)
-        Nokogiri::HTML(response_body).css('.ip div').text
+        doc = Nokogiri::HTML(response_body)
+        doc.at('h3:contains("Your Public IPv4 is: ")').text.strip.sub('Your Public IPv4 is: ', '')
```

Comment:
When the page load fails, this line now fails with "undefined method `text' for nil:NilClass" because `.at` returns either the matching `Element` or `nil`. This error is not caught by `PublicIp.get_ip` so it gets thrown all the way to callers.

Previously it would return `""`, because `.css(...)` returns a `NodeSet` which always implements `.text`.

Reply:
For me, that service returns "You have viewed this page 5 times. Your lookup limit has been reached." after calling it through `PublicIp.get_ip` a few times, which is probably why I'm getting errors here.